File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/Files2Api.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class Files2Api {
    private final HttpClient client;
    
    public Files2Api(HttpClient client) {
        this.client = client;
    }

    /** List files */
    public OpenAiFileList listFiles(Map<String, Object> params) throws Exception {
        return (OpenAiFileList) client.get(ApiPaths.aiPath("/v1/files"), params);
    }

    /** Upload file */
    public OpenAiFileObject uploadFile(UploadFilePostRequest body, Map<String, Object> params) throws Exception {
        return (OpenAiFileObject) client.post(ApiPaths.aiPath("/v1/files"), body, params, null, "multipart/form-data");
    }

    /** Retrieve file content */
    public String retrieveFileContent(String fileId) throws Exception {
        return (String) client.get(ApiPaths.aiPath("/v1/files/" + fileId + "/content"));
    }

    /** Retrieve file */
    public OpenAiFileObject retrieveFile(String fileId) throws Exception {
        return (OpenAiFileObject) client.get(ApiPaths.aiPath("/v1/files/" + fileId + ""));
    }

    /** Delete file */
    public OpenAiFileDeleteResponse deleteFile(String fileId) throws Exception {
        return (OpenAiFileDeleteResponse) client.delete(ApiPaths.aiPath("/v1/files/" + fileId + ""));
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/Files2Api.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class Files2Api(private val client: HttpClient) {

    /** List files */
    suspend fun listFiles(params: Map<String, Any>? = null): OpenAiFileList? {
        return client.get(ApiPaths.aiPath("/v1/files"), params) as? OpenAiFileList
    }

    /** Upload file */
    suspend fun uploadFile(body: UploadFilePostRequest? = null, params: Map<String, Any>? = null): OpenAiFileObject? {
        return client.post(ApiPaths.aiPath("/v1/files"), body, params, null, "multipart/form-data") as? OpenAiFileObject
    }

    /** Retrieve file content */
    suspend fun retrieveFileContent(fileId: String): String? {
        return client.get(ApiPaths.aiPath("/v1/files/$fileId/content")) as? String
    }

    /** Retrieve file */
    suspend fun retrieveFile(fileId: String): OpenAiFileObject? {
        return client.get(ApiPaths.aiPath("/v1/files/$fileId")) as? OpenAiFileObject
    }

    /** Delete file */
    suspend fun deleteFile(fileId: String): OpenAiFileDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/v1/files/$fileId")) as? OpenAiFileDeleteResponse
    }
}
